### ISSUE TITLE
Cannot find module 'eslint-config-rallycoding'

### ISSUE THREAD

**Issue by wsfuller** [AtomLinter/linter-eslint#1006]
## Bug/Question
Atom - 1.19.5
Linter - 2.2.0
Linter-eslint - 8.2.1
Node - 6.9.5

## Issue Description

    /root
      /node_modules
      package.json
      .eslintrc
      index.js

**eslintrc**

    {
      "extends": "rallycoding"
    }

**package.json**

    {
      "name": "lint-test",
      "version": "1.0.0",
      "description": "",
      "main": "index.js",
      "scripts": {
      "test": "echo \"Error: no test specified\" && exit 1"
    },
      "author": "",
      "license": "ISC",
      "devDependencies": {
        "eslint-config-rallycoding": "^3.2.0"
      }
    }

Changing `index.js` to have invalid code causes nothing to happen. No errors, warnings, identifiers absolutely no feedback is given. 

I've tried going through the [Airbnb installation](http://www.acuriousanimal.com/2016/08/14/configuring-atom-with-eslint.html) and similar results. Either there's just constant Atom Console Errors or nothing happens. Have tried both global and local installations and each attempt brings the same results and a higher level of frustration. 

## Bug Checklist

- Restarted Atom
- Disabled ALL non core plugins except Linter and Linter-eslint
- Uninstalled and reinstalled Linter and Linter-eslint
- Deleted and reinstalled `node_modules`
- Verify the `eslint` CLI gives the proper result, while `linter-eslint` does not <<<<< This isn't clear enough on how to get a Debug output
- Paste the output of the `Linter Eslint: Debug` command from the Command Palette below <<<<< This isn't clear enough on how to get a Debug output

**running `DEBUG=eslint:* eslint .`**, since no implicit commands are given in this doc returned:
```
eslint:cli Running on files +0ms
eslint:glob-util Creating list of files to process. +5ms
eslint:ignored-paths Looking for ignore file in /Users/user/development/lint-test +3ms
eslint:ignored-paths Could not find ignore file in cwd +1ms
eslint:cli-engine Processing /Users/user/development/lint-test/index.js +26ms
eslint:cli-engine Linting /Users/user/development/lint-test/index.js +1ms
eslint:config Constructing config file hierarchy for /Users/user/development/lint-test +0ms
eslint:config Using .eslintrc and package.json files +0ms
eslint:config Loading /Users/user/development/lint-test/.eslintrc +1ms
eslint:config-file Loading config file: /Users/user/development/lint-test/.eslintrc +1ms
eslint:config-file Loading rallycoding +66ms
eslint:config-file Attempting to resolve eslint-config-rallycoding +0ms
Cannot find module 'eslint-config-rallycoding'
```

**Comment by nerdgore** [comment]
I don't want to hijack this thread but this may be related to an issue I am experiencing. I am using `babel-eslint` as a parser and since the most recent update it reports it can't find the module.

```
Cannot find module 'babel-eslint' Error: Cannot find module 'babel-eslint'
    at ModuleResolver.resolve (/Users/my_username/.dotfiles/atom.symlink/packages/linter-eslint/node_modules/eslint/lib/util/module-resolver.js:74:19)
    at Object.load (/Users/my_username/.dotfiles/atom.symlink/packages/linter-eslint/node_modules/eslint/lib/config/config-file.js:554:42)
    at loadConfig (/Users/my_username/.dotfiles/atom.symlink/packages/linter-eslint/node_modules/eslint/lib/config.js:63:33)
    at getLocalConfig (/Users/my_username/.dotfiles/atom.symlink/packages/linter-eslint/node_modules/eslint/lib/config.js:130:29)
    at Config.getConfig (/Users/my_username/.dotfiles/atom.symlink/packages/linter-eslint/node_modules/eslint/lib/config.js:260:26)
    at processText (/Users/my_username/.dotfiles/atom.symlink/packages/linter-eslint/node_modules/eslint/lib/cli-engine.js:224:33)
    at CLIEngine.executeOnText (/Users/my_username/.dotfiles/atom.symlink/packages/linter-eslint/node_modules/eslint/lib/cli-engine.js:754:26)
    at lintJob (/Users/my_username/.atom/packages/linter-eslint/src/worker.js:14:20)
    at process.<anonymous> (/Users/my_username/.atom/packages/linter-eslint/src/worker.js:52:22)
    at emitTwo (events.js:111:20)
    at process.emit (events.js:191:7)
    at process.nextTick (internal/child_process.js:752:12)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
(anonymous) @ <embedded>:71186
(anonymous) @ <embedded>:71260
module.exports.Emitter.simpleDispatch @ <embedded>:63490
module.exports.Emitter.emit @ <embedded>:63606
(anonymous) @ <embedded>:71212
emitTwo @ events.js:106
emit @ events.js:191
process.nextTick @ internal/child_process.js:752
_combinedTickCallback @ internal/process/next_tick.js:67
_tickCallback @ internal/process/next_tick.js:98
/Users/my_username/.dotfiles/atom.symlink/packages/linter/lib/linter-registry.js:159 [Linter] Error running ESLint Error: Cannot find module 'babel-eslint'
    at ModuleResolver.resolve (/Users/my_username/.dotfiles/atom.symlink/packages/linter-eslint/node_modules/eslint/lib/util/module-resolver.js:74:19)
    at Object.load (/Users/my_username/.dotfiles/atom.symlink/packages/linter-eslint/node_modules/eslint/lib/config/config-file.js:554:42)
    at loadConfig (/Users/my_username/.dotfiles/atom.symlink/packages/linter-eslint/node_modules/eslint/lib/config.js:63:33)
    at getLocalConfig (/Users/my_username/.dotfiles/atom.symlink/packages/linter-eslint/node_modules/eslint/lib/config.js:130:29)
    at Config.getConfig (/Users/my_username/.dotfiles/atom.symlink/packages/linter-eslint/node_modules/eslint/lib/config.js:260:26)
    at processText (/Users/my_username/.dotfiles/atom.symlink/packages/linter-eslint/node_modules/eslint/lib/cli-engine.js:224:33)
    at CLIEngine.executeOnText (/Users/my_username/.dotfiles/atom.symlink/packages/linter-eslint/node_modules/eslint/lib/cli-engine.js:754:26)
    at lintJob (/Users/my_username/.atom/packages/linter-eslint/src/worker.js:14:20)
    at process.<anonymous> (/Users/my_username/.atom/packages/linter-eslint/src/worker.js:52:22)
    at emitTwo (events.js:111:20)
    at process.emit (events.js:191:7)
    at process.nextTick (internal/child_process.js:752:12)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
```

**Comment by Arcanemagus** [comment]
@wsfuller Sorry for the delayed response!

> Verify the `eslint` CLI gives the proper result, while `linter-eslint` does not

This is as simple as running `npx eslint .` (or however you normally run it) and checking whether it reports the same thing as what you see from `linter-eslint`. Often we get people reporting issues when `eslint` itself tells them the same thing 😉.

> Paste the output of the Linter Eslint: Debug command from the Command Palette below

The detailed steps:
1. Open the Command Palette
    * <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>P</kbd> on Windows/Linux, I'm assuming replace the <kbd>Ctrl</kbd> with <kbd>Cmd</kbd> on a Mac
    * Packages -> Command Palette -> Toggle
2. Type `Linter Eslint: Debug` into the dialog that has popped up
    * It should auto-suggest the command long before you finish typing.
3. Hit <kbd>Enter</kbd>

----

```
eslint:config-file Attempting to resolve eslint-config-rallycoding +0ms
Cannot find module 'eslint-config-rallycoding'
```

From this it looks like `eslint` on the CLI _also_ can't find the module you are attempting to use, have you forgotten to run `npm install`?

**Comment by Arcanemagus** [comment]
@nerdgore From your stack trace it looks like the package has resorted to the fallback `eslint` that is bundled with it. This is a _very_ strong indicator that you haven't installed the required modules in your project.

Please follow the above steps to get the debug information which should confirm this, and for future reference although it _looks_ like they are similar issues it's generally better to file your own issue so we can do detailed debugging without spamming other people with notifications 😉.

**Comment by IanVS** [comment]
@wsfuller did you have any luck troubleshooting this issue?  Do you still believe it is a problem in linter-eslint?

**Comment by wsfuller** [comment]
@IanVS I wasn't able to specifically identify an issue with ESLint, and was able to resolve my issue in a very non intuitive manner.

The [tl;dr](https://stackoverflow.com/questions/46126464/unable-to-get-eslint-working-in-atom-editor) is that I removed all associated Global NPM modules and did a full uninstall of Atom. From there I only added in the necessary dependencies for ESLint through Atom. My assumption is there was a conflict with JSHint (which were Atom plugins I installed previously) that may have been creating some issue. 

Right now I'm just sticking with:

* [prettier-atom](https://atom.io/packages/prettier-atom)
* [linter](https://atom.io/packages/linter)
 * [linter-ui-default](https://atom.io/packages/linter-ui-default)
* linter-eslint

Unfortunately, I haven't had time to revisit and dig into the issue too deeply then the afternoon the issues came about and spent about 30 minutes just starting over which was successful.
